pkg/pillar/cmd/zedmanager/zedmanager.go

```diff
@@ -742,3 +772,72 @@ func handleGlobalConfigDelete(ctxArg interface{}, key string,
 	*ctx.globalConfig = *types.DefaultConfigItemValueMap()
 	log.Functionf("handleGlobalConfigDelete done for %s", key)
 }
+
+func handleZedAgentStatusCreate(ctxArg interface{}, key string,
+	statusArg interface{}) {
+	handleZedAgentStatusImpl(ctxArg, key, statusArg)
+}
+
+func handleZedAgentStatusModify(ctxArg interface{}, key string,
+	statusArg interface{}, oldStatusArg interface{}) {
+	handleZedAgentStatusImpl(ctxArg, key, statusArg)
+}
+
+func handleZedAgentStatusImpl(ctxArg interface{}, key string,
+	statusArg interface{}) {
+	ctxPtr := ctxArg.(*zedmanagerContext)
+	status := statusArg.(types.ZedAgentStatus)
+	if ctxPtr.currentProfile != status.CurrentProfile {
+		ctxPtr.currentProfile = status.CurrentProfile
+		updateBasedOnProfile(ctxPtr)
+	}
+	log.Functionf("handleZedAgentStatusImpl(%s) done", key)
+}
+
+// updateBasedOnProfile check all app instances with currentProfile, set EffectiveActivate and update app instances
+func updateBasedOnProfile(ctx *zedmanagerContext) {
+	pub := ctx.subAppInstanceConfig
+	items := pub.GetAll()
+	for _, c := range items {
+		config := c.(types.AppInstanceConfig)
+		effectiveActivate := effectiveActivateCurrentProfile(config, ctx)
+		status := lookupAppInstanceStatus(ctx, config.Key())
+		if status != nil {
+			if status.EffectiveActivate != effectiveActivate {
+				log.Functionf("updateBasedOnProfile: change activate state for %s from %t to %t",
+					config.Key(), status.EffectiveActivate, effectiveActivate)
+				status.EffectiveActivate = effectiveActivate
+				changed := doUpdate(ctx, config, status)
+				if changed {
+					log.Functionf("updateBasedOnProfile: doUpdate status change for %s", status.Key())
+					publishAppInstanceStatus(ctx, status)
+				}
+			}
+		}
+	}
+}
+
+// returns effective Activate status based on Activate form app instance config and current profile
+func effectiveActivateCurrentProfile(config types.AppInstanceConfig, ctx *zedmanagerContext) bool {
```

Comment:
Since this is only using ctx.currentProfile it would be better to pass that instead of the whole zedmanagerContext.